### ISSUE TITLE
Linux Build Problem when using non out-of-source build tree

### ISSUE THREAD

**Issue by nickblock** [urho3d/Urho3D#2068]
Failed to build on ubuntu 16.04.
Latest commit 848f52f
```
CMake Error at CMake/Modules/UrhoCommon.cmake:1240 (message):
  Could not generate dependency list for PCH.  There is something wrong with
  your compiler toolchain.  Ensure its bin path is in the PATH environment
  variable or ensure CMake can find CC/CXX in your build environment.
```
g++ (Ubuntu 5.4.0-6ubuntu1~16.04.4) 5.4.0 20160609

**Comment by Mezozoysky** [comment]
I have not Ubuntu but my Debian 8 is virtually the same and i can't reproduce error with g++(4.9.2)/clang(3.5.0).
Tough, You should check your CMake/compilers versions and paths. Urho3D requires CMake >=3.5.2

**Comment by weitjong** [comment]
Just setup a new Ubuntu 16.04 VM. Not reproducible as well. Have you read our online documentation or wiki for the building steps? If you already did and still hit issue, please elaborate on how to reproduce it.

**Comment by nickblock** [comment]
Ok, well thanks for double checking for me. 
I was trying Urho3d for the first time, so just following the simplest build process, ie just fresh checkout followed by "cmake ."
My ubuntu install is pretty fresh too.

**Comment by nickblock** [comment]
Updating cmake didn't help. I also get this first:

c++: fatal error: input file ‘/XXXX/Urho3D/Source/Urho3D/Precompiled.h’ is the same as output file

**Comment by weitjong** [comment]
May be because you are not using out-of-source build tree? Like I said, please elaborate on the actual steps you have taken so that others can reproduce your issue.

**Comment by nickblock** [comment]
Yes that was it.
I had to do 

```
mkdir build
cmake ..
```

unlike your linux instructoins say.
Thanks again.

**Comment by Mezozoysky** [comment]
Please, try the following:

mkdir build #create directory
cd build #enter directory. You forgot this, as i see
cmake .. #generate makefile
make #build software
make install #optionaly install software


Sorry for my English

**Comment by nickblock** [comment]
Yes sorry, enter the build dir too!

**Comment by weitjong** [comment]
Thanks for reporting this. I think it is a bug (a regression caused by a recent commit to make our build system works better with Clion). Although non out-of-source build tree is not recommended, we still support such use case.

**Comment by slapin** [comment]
Confirming, reproduced, when build directory is not separate, bad things happen, quite recent thing.
The in-tree build is nice to have when using Urho as library, as it makes things easier.

**Comment by weitjong** [comment]
Forget to stress that although the use case is supported, it is not recommended to be used. It is not only us saying that, but also the maker of CMake and practically everyone else that knows the benefits of out-of-source build.

**Comment by nickblock** [comment]
Yeah I've never seen that before (using cmake in source like that), and it did throw me a bit! Maybe you should recommend the more usual use of cmake for your "Getting started in Linux" guide?

**Comment by weitjong** [comment]
The wiki page is maintained by the community. You can help to make it better.

**Comment by cadaver** [comment]
Think I'd like to wait for this being fixed for the 1.7 release. But certainly no hurry, I haven't had time for example to compile the changelog yet.
